tomcat-gcp-lib/src/main/java/com/google/cloud/runtimes/tomcat/session/DatastoreManager.java

```diff
@@ -0,0 +1,212 @@
+    log.debug("Datastore manager is loading session: " + id);
+    Session session = null;
+
+    try {
+      session = this.getStore().load(id);
+    } catch (ClassNotFoundException ex) {
+      log.warn("An error occurred during session deserialization", ex);
+    }
+
+    return session;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Note: Sessions are loaded at each request therefore no session is loaded at
+   * initialization.</p>
+   *
+   * @throws ClassNotFoundException Cannot occurs
+   * @throws IOException Cannot occurs
+   */
+  @Override
+  public void load() throws ClassNotFoundException, IOException {}
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Note: Sessions are persisted after each requests but never saved into the local manager,
+   * therefore no operation is needed during unload.</p>
+   *
+   * @throws IOException Cannot occurs
+   */
+  @Override
+  public void unload() throws IOException {}
+
+  /**
+   * Remove the Session from the manager but not from the Datastore
+   *
+   * @param session The session to remove.
+   */
+  @Override
+  public void removeSuper(Session session) {
+    super.remove(session);
+  }
+
+  /**
+   * Remove this Session from the active Sessions and the Datastore.
+   *
+   * @param session The session to remove.
+   */
+  @Override
+  public void remove(Session session) {
+    this.removeSuper(session);
+
+    try {
+      store.remove(session.getId());
+    } catch (IOException e) {
+      log.error("An error occurred while removing session with id: " + session.getId(), e);
+    }
+  }
+
+  /**
+   * Returns the number of session present in the Store.
+   *
+   * <p>Note: Aggregation can be slow on the Datastore, cache the result if possible</p>
+   *
+   * @return the session count.
+   */
+  @Override
+  public int getActiveSessionsFull() {
+    int sessionCount = 0;
+    try {
+      sessionCount = store.getSize();
+    } catch (IOException e) {
+      log.error("An error occurred while counting sessions: ", e);
+    }
+
+    return sessionCount;
+  }
+
+  /**
+   * Returns a set of all sessions IDs or null if an error occur.
+   *
+   * <p>Note: Listing all the keys can be slow on the Datastore.</p>
+   *
+   * @return The complete set of sessions IDs across the cluster.
+   */
+  @Override
+  public Set<String> getSessionIdsFull() {
+    Set<String> sessionsId = null;
+    try {
+      String[] keys = this.store.keys();
+      sessionsId = new HashSet<>(Arrays.asList(keys));
+    } catch (IOException e) {
+      log.error("An error occurred while listing active sessions: ", e);
+    }
+
+    return sessionsId;
+  }
+
+  @Override
+  protected void stopInternal() throws LifecycleException {
+    super.stopInternal();
+
+    if (store instanceof Lifecycle) {
+      ((Lifecycle) store).stop();
+    }
+
+    setState(LifecycleState.STOPPING);
+  }
+
+  public Store getStore() {
+    return this.store;
+  }
+
+  public void setStore(Store store) {
```

Comment:
Makes sense. Would you mind adding a short comment that explains that this method is called by the Tomcat DI framework? Otherwise it might be tough to look at this class and verify its correctness - my IDE tells me that this method is never called.